DataStructure/src/main/list/ResizableArray.java

```diff
@@ -0,0 +1,159 @@
+package main.list;
+
+/**
+ * Represents a resizable array of integer elements.
+ */
+
+public class ResizableArray {
+    private int writeIndex;
+    private int list[];
+
+    /**
+     * Constructs a resizable array.
+     * @param size
+     * size of the list.
+     */
+    public ResizableArray(int size){
+        list = new int[size] ;
+    }
+
+    /**
+     * Constructs a resizable array with initial capacity 10.
+     */
+    public ResizableArray(){
+        list = new int[10] ;
+    }
+    public void ensureCapacity(){
+        if (writeIndex == list.length){
+            int list2 [] = new int[(list.length+1)*2];
+            System.arraycopy(list,0,list2,0,list.length);
+            list = list2;
+        }
+    }
+    /**
+     * Adds an element to the array.
+     * @param num
+     */
+    public void add(int num){
+        ensureCapacity();
+        list[writeIndex] = num;
+        writeIndex++;
+    }
+
+    /**
+     * Get an element at the given index.
```

Comment:
Get -> Gets
Add java doc for return tag
who will remove ArrayIndexOutOfBoundsException ?